docs/dev_docs.md

```diff
@@ -1,3 +1,449 @@
+        '''
+        pass
+
+    def reset(self):
+        '''
+        Reset needs to initialize the `agents` attribute
+        And must set up the environment so that render(), and step()
+        can be called without issues.
+
+        Here it initializes the `num_moves` variable which counts the number of
+        hands are played.
+
+        Returns the observations for each agent
+        '''
+        self.agents = self.possible_agents[:]
+        self.num_moves = 0
+        observations = {agent: NONE for agent in self.agents}
+        return observations
+
+    def step(self, actions):
+        '''
+        step(action) takes in an action for each agent and should return the
+        - observations
+        - rewards
+        - dones
+        - infos
+        dicts where each dict looks like {agent_1: item_1, agent_2: item_2}
+        '''
+        # rewards for all agents are placed in the rewards dictionary to be ret
+        rewards = {}
+        rewards[self.agents[0]], rewards[self.agents[1]] = {
+            (ROCK, ROCK): (0, 0),
+            (ROCK, PAPER): (-1, 1),
+            (ROCK, SCISSORS): (1, -1),
+            (PAPER, ROCK): (1, -1),
+            (PAPER, PAPER): (0, 0),
+            (PAPER, SCISSORS): (-1, 1),
+            (SCISSORS, ROCK): (-1, 1),
+            (SCISSORS, PAPER): (1, -1),
+            (SCISSORS, SCISSORS): (0, 0),
+        }[(actions[self.agents[0]], actions[self.agents[1]])]
+
+        self.num_moves += 1
+        env_done = self.num_moves >= NUM_ITERS
+        dones = {agent: env_done for agent in self.agents}
+
+        # current observation is just the other player's most recent action
+        observations = {self.agents[i]: int(actions[self.agents[1 - i]]) for i in range(len(self.agents))}
+
+        # typically there won't be any information in the infos, but there must
+        # still be an entry for each agent
+        infos = {agent: {} for agent in self.agents}
+
+        return observations, rewards, dones, infos
+
+```
+
+## Wrappers
+
+A wrapper is a environment transformation which takes in an environment as input, and outputs a new environment which is similar to the inputted environment but some transformation or validation applied.
+
+### Conversion wrappers
+
+As we provide both the AEC API and the Parallel API, we also provide wrappers to convert back and forth between the two APIs.
+
+#### AEC to Parallel
+
+An environment can be converted from an AEC environment to a parallel environment with the `to_parallel` wrapper shown below. Note that this wrapper makes the following assumptions about the underlying environment:
+
+1. The environment steps in a cycle, i.e. it steps through every live agent in order.
+2. The environment does not update the observations of the agents except at the end of a cycle.
+
+Note that it also changes when rewards are allocated to agents. All rewards are given at the end of each cycle, which is slightly different than in the AEC model.
```

Comment:
```suggestion
Note that this also changes when rewards are allocated to agents. All rewards are given at the end of each cycle, which is slightly different than in the AEC model.
```
Maybe extremely briefly say what the difference is between this and the AEC model is, if possible. "Note that this also changes when rewards are given compared to the AEC API. The parallel API gives all rewards at the end of each cycle, while the AEC model give rewards..."

Reply:
done